omod/src/main/java/org/openmrs/module/idgen/rest/resource/AutoGenerationOptionResource.java

```diff
@@ -0,0 +1,245 @@
+public class AutoGenerationOptionResource extends DelegatingCrudResource<AutoGenerationOption> {
+	
+	IdentifierSourceService identifierSourceService = Context.getService(IdentifierSourceService.class);
+	
+	@Override
+	public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+		DelegatingResourceDescription description = null;
+		if (rep instanceof RefRepresentation) {
+			description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("display");
+			description.addSelfLink();
+		} else if (rep instanceof DefaultRepresentation) {
+			description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("identifierType");
+			description.addProperty("location");
+			description.addProperty("source");
+			description.addProperty("manualEntryEnabled");
+			description.addProperty("automaticGenerationEnabled");
+			description.addSelfLink();
+			description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+		} else if (rep instanceof FullRepresentation) {
+			description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("identifierType");
+			description.addProperty("location");
+			description.addProperty("source");
+			description.addProperty("manualEntryEnabled");
+			description.addProperty("automaticGenerationEnabled");
+			description.addSelfLink();
+		}
+		return description;
+	}
+	
+	@PropertyGetter("display")
+	public String getDisplayString(AutoGenerationOption autoGenerationOption) {
+		return autoGenerationOption.getIdentifierType() + " - " + autoGenerationOption.getSource() + " - "
+		        + autoGenerationOption.getLocation();
+	}
+	
+	@Override
+	public DelegatingResourceDescription getCreatableProperties() {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addRequiredProperty("identifierType");
+		description.addProperty("location");
+		description.addRequiredProperty("source");
+		description.addProperty("manualEntryEnabled");
+		description.addProperty("automaticGenerationEnabled");
+		return description;
+	}
+	
+	@Override
+	public DelegatingResourceDescription getUpdatableProperties() {
+		DelegatingResourceDescription description = getCreatableProperties();
+		description.removeProperty("identifierType");
+		return description;
+	}
+	
+	@Override
+	public SimpleObject create(SimpleObject postBody, RequestContext context) throws ResponseException {
+		Object manualEntryEnabled = (postBody.get("manualEntryEnabled"));
+		Object automaticallyEnabled = postBody.get("automaticGenerationEnabled");
+		Object sourceUuid = postBody.get("source");
+		Object locationUuid = postBody.get("location");
+		Object identifierTypeUuid = postBody.get("identifierType");
+		ArrayList<String> errors = new ArrayList<String>();
+		if ((sourceUuid == null) || (StringUtils.isBlank(sourceUuid.toString()))) {
```

Comment:
Can you remove the unnecessary brackets? When they are too many, they tend to make the code harder to read. For instance this is easier if (source == null || x == y) than if ((source == null) || (x == y))

Reply:
that makes sense doing it now